### PR TITLE
Fix nullpointer

### PR DESCRIPTION
**Pull request by moizjv**
@halkeye Reopening PR here, earlier had it in upstream.

### CODE REVIEW

File: src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java [hunk 1]
```diff
@@ -102,10 +103,24 @@ protected SauceCredentials getCredentials() {
...
+        SauceCredentials credentials = getBuildAction(build).getCredentials();
         return retrieveJobIdsFromSauce(sauceREST, build, credentials);
     }
 
+    /**
+     * @param build The build in progress
+     * @return the {@link SauceOnDemandBuildAction} instance which has been registered with the build
+     *         Can be null
+     */
+    private static SauceOnDemandBuildAction getBuildAction(Run build) {
```
Comment (moizjv) [inline src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java]: This code is repeated and copied from `SauceOnDemandReportPublisher.java`, @halkeye is it a good idea to move it into some utility file, to be reused?

File: src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java [hunk 2]
```diff
@@ -102,10 +103,24 @@ protected SauceCredentials getCredentials() {
...
+        SauceOnDemandBuildAction buildAction = build.getAction(SauceOnDemandBuildAction.class);
+        if (buildAction == null && build instanceof MavenBuild) {
+            //try the parent
+            buildAction = ((hudson.maven.MavenBuild) build).getParentBuild().getAction(SauceOnDemandBuildAction.class);
+        }
+        return buildAction;
+    }
+
```
Comment (halkeye) [inline src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java]: Okay, I see what you meant now in the other PR (why do you keep closing prs?)

Can you move this function from publisher to build action and make it static?
that way both classes can use the same function (maybe call it getSauceBuildAction or something)

Comment (moizjv) [inline src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java]: done :). Tested locally looks good. Will test it again in intrev once we merge this.

**Comment by halkeye** [comment]
👍 i like it

**Pull request merged**